includes/API/Packs/ApiLabkiPacksList.php

```diff
@@ -0,0 +1,650 @@
+	}
+
+	/**
+	 * Get a specific pack by name.
+	 *
+	 * @param \LabkiPackManager\Domain\ContentRefId $refId Ref ID
+	 * @param string $packName Pack name
+	 * @param bool $includePages Whether to include page data
+	 * @return array Array with single pack data structure
+	 */
+	private function getSinglePackByName(
+		\LabkiPackManager\Domain\ContentRefId $refId,
+		string $packName,
+		bool $includePages
+	): array {
+		$packRegistry = $this->getPackRegistry();
+		$pageRegistry = $this->getPageRegistry();
+		$repoRegistry = $this->getRepoRegistry();
+		$refRegistry = $this->getRefRegistry();
+
+		// Get the pack by name
+		$packId = $packRegistry->getPackIdByName( $refId, $packName );
+		if ( $packId === null ) {
+			$this->dieWithError(
+				'labkipackmanager-error-pack-not-found',
+				'pack_not_found'
+			);
+		}
+
+		$pack = $packRegistry->getPack( $packId );
+		if ( $pack === null ) {
+			$this->dieWithError(
+				'labkipackmanager-error-pack-not-found',
+				'pack_not_found'
+			);
+		}
+
+		// Get the ref and repo for context
+		$ref = $refRegistry->getRefById( $refId );
+		$repo = $ref ? $repoRegistry->getRepo( $ref->contentRepoId()->toInt() ) : null;
+
+		// Build and return as array
+		return [ $this->buildPackData( $pack, $repo, $ref, $pageRegistry, $includePages ) ];
+	}
+
+	/**
+	 * Build complete pack data structure.
+	 *
+	 * @param \LabkiPackManager\Domain\Pack $pack Pack domain object
+	 * @param \LabkiPackManager\Domain\ContentRepo|null $repo Repository domain object
+	 * @param \LabkiPackManager\Domain\ContentRef|null $ref Ref domain object
+	 * @param \LabkiPackManager\Services\LabkiPageRegistry $pageRegistry Page registry
+	 * @param bool $includePages Whether to include page data
+	 * @return array Complete pack data structure
+	 */
+	private function buildPackData(
+		\LabkiPackManager\Domain\Pack $pack,
+		?\LabkiPackManager\Domain\ContentRepo $repo,
+		?\LabkiPackManager\Domain\ContentRef $ref,
+		\LabkiPackManager\Services\LabkiPageRegistry $pageRegistry,
+		bool $includePages
+	): array {
+		// Count pages for this pack
+		$pageCount = $pageRegistry->countPagesByPack( $pack->id() );
+
+		$data = [
+			'pack_id' => $pack->id()->toInt(),
+			'content_ref_id' => $pack->contentRefId()->toInt(),
+			'name' => $pack->name(),
+			'version' => $pack->version(),
+			'source_commit' => $pack->sourceCommit(),
+			'installed_at' => $pack->installedAt(),
+			'installed_by' => $pack->installedBy(),
+			'updated_at' => $pack->updatedAt(),
+			'status' => $pack->status(),
+			'page_count' => $pageCount,
+		];
+
+		// Add repo and ref context if available
+		if ( $repo !== null ) {
+			$data['repo_id'] = $repo->id()->toInt();
+			$data['repo_url'] = $repo->url();
+		}
+
+		if ( $ref !== null ) {
+			$data['ref'] = $ref->sourceRef();
+			$data['ref_name'] = $ref->refName();
+		}
+
+		// Include pages if requested
+		if ( $includePages ) {
+			$pages = $pageRegistry->listPagesByPack( $pack->id() );
```

Comment:
this seems like it should be another endpoint unless the DB system has some way of doing an efficient `SELECT IN` to get joined rows